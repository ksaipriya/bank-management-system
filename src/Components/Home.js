import React from 'react'
import { useAuth0 } from '@auth0/auth0-react'
import { Form, Button, Col, Container } from 'react-bootstrap';
import 'react-datepicker/dist/react-datepicker.css'
import { Label } from 'reactstrap';

const Home = () => {
    const { loginWithRedirect, isAuthenticated } = useAuth0()
    // landing page
    return (
        <div>
            {!isAuthenticated &&
                <Container>
                    <h4 data-testid="header">Welcome to XXX Bank</h4>
                    <Form>
                        <Form.Row>
                            <Form.Group as={Col} className="register">
                                <Label>Please Authenticate your email and register to enjoy our services</Label>
                            </Form.Group>
                            {/*authenticate button redirects to auth0 website*/}
                            <Form.Group as={Col} className="register">
                                <Button onClick={() => loginWithRedirect()}>Authenticate</Button>
                            </Form.Group>
                        </Form.Row>
                    </Form>
                </Container>
            }
        </div>
    )
}
export default Home
