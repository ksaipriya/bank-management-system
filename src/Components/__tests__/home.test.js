import React from 'react';
import { render } from '@testing-library/react';
import Home from '../Home.js'
import ShallowRenderer from 'react-test-renderer/shallow';
import Adapter from 'enzyme-adapter-react-16';
import * as enzyme from 'enzyme';

enzyme.configure({ adapter: new Adapter() });
describe('Test case for Home Component', () => {
    it('matches snapshot', () => {
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(<Home
            x={true}
        />);
        const result = shallowRenderer.getRenderOutput();
        expect(result).toMatchSnapshot();

    });

    it('check title', () => {
        let wrapper = render(<Home x={false} />)
        const title = wrapper.getByTestId('header')
        expect(title.textContent).toBe('Welcome to XXX Bank')
    })


    it ('authenticate label', () => {
        const { getByText } = render(<Home x={true} />)
        const name = getByText('Please Authenticate your email and register to enjoy our services');
        expect(name).toBeInTheDocument();
    });

    it('Test click event', () => {
        const mockCallBack = jest.fn();
        const button = enzyme.shallow(<button onClick={mockCallBack}>Authenticate</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });
})