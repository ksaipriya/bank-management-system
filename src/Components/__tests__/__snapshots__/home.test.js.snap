// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for Home Component matches snapshot 1`] = `
<div>
  <Container
    fluid={false}
  >
    <h4
      data-testid="header"
    >
      Welcome to XXX Bank
    </h4>
    <Form
      inline={false}
    >
      <FormRow>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          className="register"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Please Authenticate your email and register to enjoy our services
          </Label>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          className="register"
        >
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            variant="primary"
          >
            Authenticate
          </Button>
        </FormGroup>
      </FormRow>
    </Form>
  </Container>
</div>
`;
